src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs

```diff
@@ -59,17 +60,24 @@ public async Task UploadErrorsAsync(string[] importErrors, CancellationToken can
                     await writer.WriteLineAsync(error);
                 }
 
+#pragma warning disable CA2016 // .NET 6 does not have cancel. Remove when we switch to .NET8
                 await writer.FlushAsync();
                 stream.Position = 0;
 
                 string blockId = Convert.ToBase64String(Guid.NewGuid().ToByteArray());
                 await _integrationDataStoreClient.UploadBlockAsync(_fileUri, stream, blockId, cancellationToken);
-                await _integrationDataStoreClient.AppendCommitAsync(_fileUri, new string[] { blockId }, cancellationToken);
+                await _integrationDataStoreClient.AppendCommitAsync(_fileUri, [blockId], cancellationToken);
             }
-            catch (Exception ex)
+            catch (IntegrationDataStoreException ex)
             {
                 _logger.LogWarning(ex, "Failed to upload import error log.");
-                throw new RetriableJobException(ex.Message, ex);
+                if (retries++ < 3)
+                {
+                    await Task.Delay(5000, cancellationToken);
+                    goto retry;
+                }
+
+                throw;
```

Comment:
We should use Polly for retries going forward.

Reply:
Will fix